owncloudApp/src/androidTest/java/com/owncloud/android/flow/CreatePublicShareTest.kt

```diff
@@ -0,0 +1,370 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.flow
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.content.Context
+import android.content.Intent
+import androidx.lifecycle.MutableLiveData
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import com.owncloud.android.R
+import com.owncloud.android.authentication.AccountAuthenticator.KEY_AUTH_TOKEN_TYPE
+import com.owncloud.android.data.DataResult
+import com.owncloud.android.data.capabilities.db.OCCapabilityEntity
+import com.owncloud.android.data.sharing.shares.db.OCShareEntity
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.lib.common.accounts.AccountUtils
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.status.CapabilityBooleanType
+import com.owncloud.android.lib.resources.status.OwnCloudVersion
+import com.owncloud.android.presentation.viewmodels.capabilities.OCCapabilityViewModel
+import com.owncloud.android.presentation.viewmodels.sharing.OCShareViewModel
+import com.owncloud.android.presentation.ui.sharing.ShareActivity
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.utils.AccountsManager
+import com.owncloud.android.utils.AppTestUtil
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.koin.android.ext.koin.androidContext
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
+
+class CreatePublicShareTest {
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(
+        ShareActivity::class.java,
+        true,
+        false
+    )
+
+    private lateinit var file: OCFile
+
+    private val publicShares = arrayListOf(
```

Comment:
why do you use arrayList?? It's always better to interfaces, in this case List ;-)